app/scripts/background.js

```diff
@@ -333,6 +335,32 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+/**
+ * Overrides the Content-Security-Policy (CSP) header by adding a nonce to the `script-src` directive.
+ * This is a workaround for [Bug #1446231](https://bugzilla.mozilla.org/show_bug.cgi?id=1446231),
+ * which involves overriding the page CSP for inline script nodes injected by extension content scripts.
+ */
+function overrideContentSecurityPolicyHeader() {
+  // The extension url is unique per install on Firefox, so we can safely add it as a nonce to the CSP header
+  const nonce = btoa(browser.runtime.getURL('/'));
+  browser.webRequest.onHeadersReceived.addListener(
+    ({ responseHeaders, url }) => {
+      // Check whether inpage.js is going to be injected into the page or not.
+      // There is no reason to modify the headers if we are not injecting inpage.js.
+      if (checkURLForProviderInjection(new URL(url))) {
```

Comment:
nit: naming of this function could be improved per our developer guidelines on function names: https://github.com/MetaMask/contributor-docs/blob/main/docs/javascript.md#use-a-verb-phrase-for-functionmethod-names